Added https support for prefixes uploaded by url

Shamanou opened the pull request:
Added https support for prefixes uploaded by url as well as cleaned up the file

Shamanou committed:
Added https support for prefixes uploaded by url

kburger commented on extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java:
```diff
@@ -156,20 +152,46 @@ private String parseFullFile(FileItem file, String baseuri) {
+            con = (HttpsURLConnection) obj.openConnection();
+        } else {
+            con = (HttpURLConnection) obj.openConnection();
+        }
```
This can be improved. Either `obj.getProtocol().equals("https")` (or `equalsIgnoreCase`), or `url.charAt(4) == 's'`. But then again, why cast it to `HttpsURLConnection` in the first place? [`HttpsURLConnection`](https://docs.oracle.com/javase/8/docs/api/index.html?javax/net/ssl/HttpsURLConnection.html) is a subclass of `HttpURLConnection`. You initialize a `HttpURLConnection` declared object, so the specifics of the `Https` class is not used anyway.

kburger commented on extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java:
```diff
@@ -156,20 +152,46 @@ private String parseFullFile(FileItem file, String baseuri) {
+                        continue;
+                    } catch (RDFHandlerException e) {
+                        continue;
+                    }
```
If you don't handle the specific exceptions, this should be a multi-catch.

kburger reviewed (changes requested):
Some basic logic should be changed, in my opinion.

Shamanou committed:
Imrpoved file format parsing when uploading from prefix from url

The baseuri problem is solved!
Now prefixes can be loaded from any uri without having to have knowledge of the base uri in the graph.

Some rdf4j trickery did this, main part is in the vocabularyimporter.  Also migrated from any23 to rdf4j and updated code style.

Shamanou committed:
Updated prefix parsing

Did some edits after testing, some extra conditions where needed to let files without a base uri in the prefixes be accepted, also accept headers are concatinated and not replaced

Shamanou committed:
Added nescessary files to build.xml

Shamanou committed:
Improved file prefix from url parsing

cleaned up code
improved accept headers

Shamanou committed:
removed unneeded comment

Shamanou commented:
I made the parsing of prefixes more robust as well

Shamanou commented on extensions/fairifier-rdf-extension/dtls/fairifier/DetectFileFormatCommand.java:
```diff
@@ -155,22 +133,4 @@ private String parseFullFile(FileItem file, String baseuri) {
 
     }
-
-    private String getFormatFromUrl(String url)
```
This was unused so i removed it, the backend handles the parsing from url in a different call.
The method was moved to where it was needed

Shamanou commented on extensions/fairifier-rdf-extension/deri/grefine/rdf/vocab/VocabularyImporter.java:
```diff
@@ -23,171 +25,248 @@
+    }
+
+    private RDFParser getParserFromUrl(String url) throws IOException, MalformedURLException {
+        StringBuffer response = new StringBuffer();
```
This method is new, it returns a parser of the format that is either returned by the content header of the request or by parsing the content itself full parsing is only done if a plain_text header is returned

Shamanou commented on extensions/fairifier-rdf-extension/deri/grefine/rdf/vocab/VocabularyImporter.java:
```diff
@@ -23,171 +25,248 @@
+    }
+
+    private Model getModelFromUrl(String url)
+            throws IOException, MalformedURLException, IOException, ProtocolException {
```
This method is new, it uses the information of the parser that fits the url to retrieve and parse the ontology correctly

Shamanou committed:
Style updates

Shamanou committed:
pypi link of virtual env didnt work so i updated it

Shamanou committed:
Fixed tests

Testing didnt work to fix this i added some longer waiting times for prexis loading so that after the ajaxelement closes it leaves some time for the actual loading

Shamanou committed:
Updated file with PR commentary

Moved parsers and other static variables to the top.
Added check for optional used in parsing. Added multicatch instead of catch 'list' .

Used string builder to contatinate acceptheader and trimmed the final ',' from the acceptheader

Shamanou commented:
I also have to update the refine script, the pypi link for virtualenv did not work anymore

Shamanou committed:
undid typo

Shamanou committed:
Https should be enabled now

made changes to new prefix widget to not block https prefixes

kburger commented on main/tests/client/src/facets.js:
```diff
@@ -130,15 +130,41 @@ var test_facets = new function() {
   });
   wait(test, "forAjaxEnd");
+  wait(test, "forPageLoad", {
+    timeout: "50000"
```
Adding an arbitrary timeout might not be the cleanest solution. Are there alternatives? Like waiting for a certain element to show or disappear?

Shamanou replied:
i looked for that but windmill has a very limited scope in wait commands, there is a waitforelement, but the elements are already loaded so that does not apply here

Shamanou commented on extensions/fairifier-rdf-extension/scripts/rdf-schema-new-prefix-widget.js:
```diff
@@ -1,151 +1,146 @@
-    		return;
-    	}     	
-        console.log(uri)
-        if (uri.split(":")[0].toLowerCase() === "https"){
```
This is what is removed, the rest is style updates

kburger reviewed (approved):
Minor comment, approved for the rest of it.
